vm-setup/roles/v1a2_integration_test/templates/controlplane_centos.yaml

```diff
@@ -0,0 +1,107 @@
+apiVersion: cluster.x-k8s.io/v1alpha2
+kind: Machine
+metadata:
+  name: {{ CLUSTER_NAME }}-controlplane-0
+  labels:
+    cluster.x-k8s.io/control-plane: "true"
+    cluster.x-k8s.io/cluster-name: "{{ CLUSTER_NAME }}"
+spec:
+  version: {{ KUBERNETES_VERSION }}
+  bootstrap:
+    configRef:
+      apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+      kind: KubeadmConfig
+      name: {{ CLUSTER_NAME }}-controlplane-0
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+    kind: BareMetalMachine
+    name: {{ CLUSTER_NAME }}-controlplane-0
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+kind: BareMetalMachine
+metadata:
+  name: {{ CLUSTER_NAME }}-controlplane-0
+spec:
+  image:
+    url: {{ IMAGE_URL }}
+    checksum: {{ IMAGE_CHECKSUM }}
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+kind: KubeadmConfig
+metadata:
+  name: {{ CLUSTER_NAME }}-controlplane-0
+spec:
+  initConfiguration:
+    nodeRegistration:
+      name: {{ " '{{ ds.meta_data.name }}' " }}
+      kubeletExtraArgs:
+        node-labels: 'metal3.io/uuid={{ '{{ ds.meta_data.uuid }}' }}'
+  preKubeadmCommands:
+    - ifup eth1
+    - yum check-update
+    - yum install -y gcc kernel-headers kernel-devel
+    - yum update -y
+    - yum install -y keepalived
+    - systemctl start keepalived
+    - systemctl enable keepalived
+    - yum install -y yum-utils device-mapper-persistent-data lvm2
+    - yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+    - yum install docker-ce-18.09.9 docker-ce-cli-18.09.9 containerd.io -y
+    - usermod -aG docker centos
+    - systemctl start docker
+    - systemctl enable docker
+    - setenforce 0
+    - sed -i 's/^SELINUX=enforcing$/SELINUX=permissive/' /etc/selinux/config
+    - yum install -y kubelet kubeadm kubectl --disableexcludes=kubernetes
+    - systemctl enable --now kubelet
+  postKubeadmCommands:
+    - mkdir -p /home/centos/.kube
+    - cp /etc/kubernetes/admin.conf /home/centos/.kube/config
+    - chown centos:centos /home/centos/.kube/config
+  files:
+    - path: /etc/keepalived/keepalived.conf
+      content: |
+        ! Configuration File for keepalived
+        global_defs {
+            notification_email {
+            sysadmin@example.com
+            support@example.com
+            }
+            notification_email_from lb@example.com
+            smtp_server localhost
+            smtp_connect_timeout 30
+        }
+        vrrp_instance VI_1 {
+            state MASTER
+            interface eth1
+            virtual_router_id 1
+            priority 101
+            advert_int 1
+            virtual_ipaddress {
+                192.168.111.249
```

Comment:
Same should be derived from baremetal_network_cidr although I see the crs/v1alpha2/* already has this hardcoded in a few places.

Reply:
Yes, those crs are copied from crs/v1alpha2/ with only variable changes for Jinja2 templating.